### DISCUSSION TITLE
message: Access to requested resource is denied, code: Unauthorized

### DISCUSSION THREAD

**Discussion by od081274gil** [amzn/selling-partner-api-models#3495] [General]
Hi 
I have problem, who can halp me ?
We have a published application with users, for some users sometimes we get the error "Access to requested resource is denied" for all api query"
Example 
Date: Tue, 31 Jan 2023 15:13:26 GMT
Content-Type: application/json
Content-Length: 141
Connection: keep-alive
x-amzn-RequestId: a915abe5-33e0-4bd5-a6e9-45ae0033bedd
x-amzn-ErrorType: AccessDeniedException

  "errors": [
    {
      "message": "Access to requested resource is denied.",
     "code": "Unauthorized",
     "details": ""
    }
  ]
}

**Comment by born2net** [comment]
same issue, any resolutions?

**Comment by MeenaAmz** [comment]
I believe by users you mean sellers. Have those sellers authorized your app to access their data? If yes, is this happening sporadically or all the time for the few users? When you get unauthorized it usually means that they don't have access.

**Comment by born2net** [comment]
but in my setup I am just trying to access my own application, my own products and catalog.
Do we need to wait for Amazon to approve the application? it has been over a week now.
I spent every possible configuration humanly possible.

**Reply by MeenaAmz** [reply]
Are you using self authorization workflow for your app (https://developer-docs.amazon.com/sp-api/docs/self-authorization)? Access depends on how you are authorizing your application. Once you have the required LWA access token, make sure that you have followed all the steps for connecting to sp api (https://developer-docs.amazon.com/sp-api/docs/connecting-to-the-selling-partner-api)

**Reply by born2net** [reply]
how do I specify us-east-1 (instead of 'na' north america) , maybe that's the issue

**Reply by sungolivia** [reply]
there is a step with a screenshot where you can specify the region
https://developer-docs.amazon.com/sp-api-blog/docs/how-to-test-selling-partner-api-endpoints

**Comment by born2net** [comment]
this is how I call it with zero luck:

![image](https://user-images.githubusercontent.com/5090936/223762085-e0c1e9f3-26fd-45d2-badf-cf3238dc2d60.png)

**Reply by born2net** [reply]
the fact I get message: 'Access to requested resource is denied.' telling me that all the credentials are correct (if I change any credentials data the error is "The security token included in the request is invalid" so I know that all my credentials are correct, I played with the role and user in IAM for soooo long... but can't get pass the 'Access to requested resource is denied.' error. I am so frustrated.  Even the refresh token is correct ... so what am I doing wrong :(

**Comment by born2net** [comment]
I also have to say this is the only command that works for me:

```
let res = await sellingPartner.callAPI({
            operation:'getMarketplaceParticipations',
            endpoint:'sellers'
        });
        console.log(res)
```
